src/client/camera.cpp

```diff
@@ -454,7 +454,19 @@ void Camera::update(LocalPlayer* player, f32 frametime, f32 busytime, f32 tool_r
 	if (player->getPlayerControl().zoom && player->getZoomFOV() > 0.001f) {
 		fov_degrees = player->getZoomFOV();
 	} else {
-		fov_degrees = m_cache_fov;
+		// Apply server-sent FOV. If server doesn't
+		// enforce client's FOV, default to m_cache_fov
+		auto fov_spec = player->getFov();
+		if (fov_spec.first > 0.0f) {
+			// If server-sent FOV is a multiplier, multiply
+			// it with m_cache_fov instead of overriding
+			if (fov_spec.second)
+				fov_degrees = m_cache_fov * fov_spec.first;
+			else
+				fov_degrees = fov_spec.first;
+		} else {
+			fov_degrees = m_cache_fov;
```

Comment:
Shouldn't this apply with the zoom key pressed, too?

Reply:
I'm not sure how set_fov would work with the existing zoom functionality, which is why I left it as it is (in hindsight, probably not the best thing to do)

- If server sends an FOV override, **and** player presses zoom key, which one overrides the other?
- If server-sent FOV is a multiplier, what would happen then? Doubling the existing zoom, for example, would be messed up.

Reply:
> If server sends an FOV override, and player presses zoom key, which one overrides the other?

the server

> If server-sent FOV is a multiplier, what would happen then? Doubling the existing zoom, for example, would be messed up.

no issue with that, the fov is still clamped to a sane range

Reply:
> > If server sends an FOV override, and player presses zoom key, which one overrides the other?
> 
> the server

Uh... ok

> > If server-sent FOV is a multiplier, what would happen then? Doubling the existing zoom, for example, would be messed up.
> 
> no issue with that, the fov is still clamped to a sane range

FOV is clamped between `1.0f` and `160.0f`. Sane? Sure. Playable? Nope.

But if the server has the authority to override client-side zoom, then this isn't a problem. Thanks for clarifying.

Reply:
>But if the server has the authority to override client-side zoom, then this isn't a problem. 

Yes, that was the idea here.
If the server does stupid things that impact zoom functionality or make the game unplayable it's not our problem. The server has full control and full responsibility.

Reply:
I've juggled the code around a bit to give priority to server-sent overrides first, then client-side zoom, and then default to client's preferred FOV:

```cpp
f32 fov_degrees;
auto fov_spec = player->getFov();
if (fov_spec.first > 0.0f) {
	// If server-sent FOV is a multiplier, multiply
	// it with m_cache_fov instead of overriding
	if (fov_spec.second)
		fov_degrees = m_cache_fov * fov_spec.first;
	else
		fov_degrees = fov_spec.first;
} else if (player->getPlayerControl().zoom && player->getZoomFOV() > 0.001f) {
	// Player requests zoom, apply zoom FOV
	fov_degrees = player->getZoomFOV();
} else {
	// Set to client's selected FOV
	fov_degrees = m_cache_fov;
}
```

Now, zoom is only checked if the server-sent FOV is `0.0f`.